### PR TITLE
Make TimeDerivativeNodalKernel inherit from TimeNodalKernel

### LINKED ISSUE #5883: Time Integration with NodalKernels
NodalKernels were introduced in #5815 but they show issues with high order time integration.  We need to get that figured out.

### LINKED ISSUE #5949: TimeNodalKernel not tested
I see a test for this but it doesn't appear to get hit:

http://mooseframework.org/docs/coverage/moose/coverage/src/nodalkernels/TimeNodalKernel.C.gcov.html

ping @friedmud

### PR DESCRIPTION
**Pull request by friedmud**
Whoops!

At least I get to close two tickets at once!

Thanks @jwpeterson and @permcody for helping on this one!

closes #5883 closes #5949

### CODE REVIEW

**Comment by jwpeterson** [comment]
Looks like this would be fixed by regolding maybe?  Or is there something deeper going on here?

**Comment by friedmud** [comment]
I didn't even know this was still here.  I'll try to get to it soon....

**Comment by permcody** [comment]
Looking back through old open PRs: This one really needs to go in. This outstanding bug is pretty serious.  Well serious if somebody tried to use it anyway.

**Comment by friedmud** [comment]
Yeah - I'm not sure why it failed.  I don't have time to look at this until next week though.  But I _will_ definitely have time next week.

Sorry for the delay... just completely destroyed right now.

**Comment by permcody** [comment]
ping @friedmud

**Comment by friedmud** [comment]
Gotcha.  Both of these will be fixed today.

**Comment by friedmud** [comment]
sonofa

**Comment by friedmud** [comment]
@permcody @jwpeterson  I can't explain these test results.  For some reason I get different results for the higher order test on Linux and OSX.  I've been putzing around with it all afternoon.  I'm freaking done for today.

If one of you wouldn't mind building this branch for me and just verifying what I'm seeing I would appreciate it...

**Comment by friedmud** [comment]
BTW: I ran it through valgrind... it didn't show anything...

**Comment by permcody** [comment]
I just built this branch on both Rod and my workstation and all tests are passing for me on both platforms. Which platform are you seeing the failure on?

**Comment by permcody** [comment]
I should say, which one is diffing against the current gold file?

**Comment by friedmud** [comment]
Ok  - let me try again.  As it's currently checked in it was passing on Linux (and failing on my laptop).  But let me try again (I could have had some branch wires crossed ;-)

**Comment by friedmud** [comment]
( @permcody Thanks for checking it... this is why I didn't want to continue to beat my head against the wall until someone else tried! )

**Comment by permcody** [comment]
It is definitely passing for me on my workstation as-is in serial, parallel, threaded and recover modes.

**Comment by friedmud** [comment]
Weird - it is still failing for me on my laptop!

Can we get a third opinion?

**Comment by permcody** [comment]
On Mon, Jan 4, 2016 at 2:38 PM Derek Gaston <notifications@github.com>
wrote:

> Weird - it is still failing for me on my laptop!
>
> Can we get a third opinion?
>

Sure, you're still CrAzY :p

I just ran the same branch on my laptop and it works there too. Perhaps
it's a PETSc issue or some other environment factor? I have consistent
environments on both my laptop and workstation.

**Comment by jwpeterson** [comment]
On Mon, Jan 4, 2016 at 2:38 PM, Derek Gaston <notifications@github.com>
wrote:

> Weird - it is still failing for me on my laptop!
>
> Can we get a third opinion?
>

Have you updated and/or not updated your laptop recently?

-- 
John

**Comment by friedmud** [comment]
If you guys think it's good.  Then merge away.  No idea what's up with my junk...
